recipes/security_realm_file.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Cookbook Name:: cdap
+# Recipe:: security_realm_file
+#
+# Copyright © 2013-2015 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Manage Authentication realmfile
+realmfile = node['cdap']['cdap_site']['security.authentication.basic.realmfile']
+realmdir = ::File.dirname(realmfile)
+
+# Ensure parent directory exists
+directory realmdir do
+  action :create
+  recursive true
```

Comment:
Set owner and group to cdap and mode to `'0700'` for security *unless* this directory is *also* CDAP's conf dir, since non-CDAP processes need to read `cdap-site.xml`, like CLI.